modules/openapi-generator/src/main/resources/php-dt/model_object.mustache

```diff
@@ -16,9 +16,16 @@ class {{classname}}
 }}{{^internal.fromParameters}}{{>model_normal_var}}{{/internal.fromParameters
 }}{{/vendorExtensions
 }}{{^vendorExtensions}}{{>model_normal_var}}{{/vendorExtensions
-}}     * @var {{dataType}}|null
+}}     * @var {{^isReferenceToEnum}}{{dataType}}{{/isReferenceToEnum}}{{#isReferenceToEnum}}{{referencedEnumDataType}}{{/isReferenceToEnum}}{{^required}}|null{{/required}}
```

Comment:
Got your point, but would it be nicer if we use the default value for these fields instead? Since by OpenAPI spec, they can never be null

Reply:
> if we use the default value for these fields instead

Sure, if default value is declared in specification, it will be great to initialize property with this value and then there is little (or no) need to allow `null` in DocBlock. But it is hard to imagine required field with default value because if field is marked required it is usually impossible to provide meaningful default value.

> Since by OpenAPI spec, they can never be null

I strongly disagree here - it only means that they can not be null after successful hydration (that is ensured with DataTransfer annotations set for property). So such property can be null unless it is initialized in its declaration or in class constructor. If you worked with Doctrine ORM the situation is similar to autogenerated id fields - every entity object should have an id but new entity gets one only after DB commit.

Reply:
The hydration process is nothing related to OpenAPI spec, that's another challenge per each language, after reading more thoroughly the open api specs, I can tell the optional/required is not a good indication for nullability, but the nullable attribute (https://swagger.io/specification/#schema-nullable) is solely for this purpose. Anyway because of the situation with current PHP implementation, I adjusted the code so that it will use default values for fields which either has the default value defined or we can know them ourself (for basic types, like 0 for number or integer, or empty string for string type). But the code now marks the field also as php nullable if it is not nullable by open api specs but we can't know the default value and it is not specified in the open api spec file. Please do check the generated sample files.

Reply:
Please reread previous comments - you are going in the wrong direction.

OpenAPI specification describes **data that API expects to receive (1)** and **data that API promises to return (2)**. It describes neither **data structure for API request preparation (3)** nor **data structure for API response parsing (4)**. It is like caring about the flavour of the tea you drink, not the shape of the cup you are drinking this tea from (if you allow me such analogy).

And the code generated with `php-dt` provides **(3)** and **(4)** and presumes the following usage:
* you create an empty **(3)**
* you fill it with valid **(1)** somehow
* you call generated API client with filled **(3)** 
* API client uses filled **(3)** to acquire **(2)**
* API client ensures that acquired **(2)** is valid according API spec
* API client creates empty **(4)**
* API client fills this empty **(4)** with acquired **(2)**
* you receive filled **(4)** from API client

So:
* it is completely "ok" that field from **(1)** is required and yet corresponding field from **(3)** is nullable
* it is completely "ok" that field from **(2)** is required and yet corresponding field from **(4)** is nullable
* providing default values for required fields (that are not declared in API spec) is harmful because it breaks validation (no error is raised if API forgets to return required field)

And by the way default value calculation [here](https://github.com/OpenAPITools/openapi-generator/pull/11533/commits/ea821c6a9b4886a52c9ac709b2049b49884d4295) does not provide valid values if there are additional restrictions like pattern for string or minimum amount of items for collection.

Reply:
Ok, at first, I anticipated the php-dt as php data-transfer as a general term and your lib just as an extra validator, but as I read your lib page I understand your intentions, though I don't agree with them. Maybe it's better for me to target my changes towards php-laravel.
I do not agree with your analogy and the method of implementation you mentioned above. Let me ask a question, why don't you use `mixed` for all types then and bothering using types? That way, It will be even more loose, and the variables can be anything. 
About the default values in complex scenarios with validations, it would be the responsibility of the one who defined the API (and OpenAPI spec) to provide a correct default value (and we will use it).

In general, strict typing helps prevent lots of errors and issues, but this is your call and I appreciate it.

Reply:
> Maybe it's better for me to target my changes towards php-laravel.

Oh, so that seems the root of misunderstanding. `php-dt` is client generator - it provides library for making requests to the existing API described by OpenAPI specification. `php-laravel` is server generator - it provides code stub for implementing the API described by OpenAPI specification.

> Let me ask a question, why don't you use mixed for all types then and bothering using types? That way, It will be even more loose, and the variables can be anything.

Guess I am terrible at explaining... The goal was not to make something loose, the goal was to provide correct strict typing description for generated data structures.